### DISCUSSION TITLE
/sell all - protection for named items or items with enchant

### DISCUSSION THREAD

**Discussion by bobhenl** [Zrips/CMI#4278] [Feature Requests]
**Description of feature/improvement request:**

---
### Detailed explanation of what it should do, when and in what situation: ###
Hi, could you please add way to set protection for /sell all command? Because now, when players are mining for ex. or searching treasures and they have armor etc., and there'S worth set for it due to for ex. golden and iron tools and armors can be found in treasures and aren't mostly used... that¨s why it's in sell gui, then when players will use /sell all, their armor will be sold. The problem is when they have on this golden or iron armor or tools good enchantments OR it's special item, like Slimefun item or just special by name and lore. So useful will be if you add way to disable selling items with nbt tags maybe? Or with name, lore, etc.?

**Comment by CloudeLecaw** [comment]
i recomment to create a custom alias instead of using the all argument

> sell (all/blocks/hand/same/gui)

```
  sellallblocks:
    Cmds:
    - cmi sell block
```
just sells all blocks - these normaly are not named

for consume and otheritems i recomment to let them use the gui.
or you simply let your members use /sell gui that opens a gui where they can sell all they want. if they now sell a named items its their own fault

permissions: cmi.command.sell.[blocks/all|same|gui]

i know /sell all it a the lazzynut way for player tofast sell your items.
but the command does in that case exacly what it should. it sells ALL. so there is nothing wrog in its workage.

so - dont let your player use sell all if u want this seperated from what it does
